### DISCUSSION TITLE
Column Pinning -- Cannot see all columns when width of grid does not allow for all of the pinned columns to be displayed

### DISCUSSION THREAD

**Discussion by romangit12** [IgniteUI/igniteui-angular#14023] [Q&A] [answered]
Hi Infragistics, 

I am using Ignite UI for Angular and have an issue concerning the column pinning functionality. I see in my project and in the Infragistics documentation examples that only unpinned columns are able to scrolled through horizontally.

Although this is working as expected, I find that when all of the columns are pinned and the width of the grid does not allow for all of the pinned columns to be displayed, the grid does not have a scrollbar in the grid allowing for the user to see all of the pinned columns. This means that only the first few selected columns are able to be viewed, and the rest of the columns will not be able to be viewed. I see this is happening for the examples in the documentation as well (https://www.infragistics.com/products/ignite-ui-angular/angular/components/grid/column-pinning).

In my project, sometimes the grid will expand horizontally and a horizontal window scrollbar appears, allowing the user to scroll and view all of the columns. However, this is also not ideal, as sometimes the screen is extremely elongated when there is a large number of pinned columns.

Is there any solution or workaround for the pinning of a large number of columns in a grid?

Thank you in advance

**Comment by ChronosSF** [comment, marked as answer]
Hello @romangit12 ,

This behavior is by design. The idea of pinning columns is to have a few important columns always visible and not part of the horizontal scroll . Having two horizontal scrollbars for both areas or one single one scrolling both defeat the purpose of pinning in the first place. We decided against putting arbitrary limitations to how many columns can be pinned as this scenario is best handled on an application level, whether by disallowing pinning specific columns or preventing pinning after a certain count, or anything else that makes sense for the application.
